### PR TITLE
chore: reduce size of auth modules 

### PR DESCRIPTION
**Pull request by williazz**
Small refactor to share logic between BasicAuthentication and EnhancedAuthentication.

### 📊 Package size report&nbsp;&nbsp;&nbsp;<kbd>-0.56%↓</kbd>

| File                                                |     Before |                        After |
| --------------------------------------------------- | ---------: | ---------------------------: |
| `dist/cjs/dispatch/Authentication.d.ts`             |   `3.2 kB` |       <sup>5%↑</sup>`3.3 kB` |
| `dist/cjs/dispatch/Authentication.js`               |  `10.9 kB` |   <sup>-18.3%↓</sup>`8.9 kB` |
| `dist/cjs/dispatch/BasicAuthentication.d.ts`        |          — |                      `682 B` |
| `dist/cjs/dispatch/BasicAuthentication.js`          |          — |                     `5.9 kB` |
| `dist/cjs/dispatch/EnhancedAuthentication.d.ts`     |   `3.2 kB` |    <sup>-79.4%↓</sup>`655 B` |
| `dist/cjs/dispatch/EnhancedAuthentication.js`       |  `10.3 kB` |   <sup>-49.8%↓</sup>`5.2 kB` |
| `dist/cjs/orchestration/Orchestration.js`           |  `14.9 kB` |    <sup>0.1%↑</sup>`14.9 kB` |
| `dist/es/dispatch/Authentication.d.ts`              |   `3.2 kB` |       <sup>5%↑</sup>`3.3 kB` |
| `dist/es/dispatch/Authentication.js`                |  `10.7 kB` |   <sup>-18.3%↓</sup>`8.7 kB` |
| `dist/es/dispatch/BasicAuthentication.d.ts`         |          — |                      `682 B` |
| `dist/es/dispatch/BasicAuthentication.js`           |          — |                     `5.7 kB` |
| `dist/es/dispatch/EnhancedAuthentication.d.ts`      |   `3.2 kB` |    <sup>-79.4%↓</sup>`655 B` |
| `dist/es/dispatch/EnhancedAuthentication.js`        |  `10.1 kB` |   <sup>-50.4%↓</sup>`5.0 kB` |
| `dist/es/orchestration/Orchestration.js`            |  `14.2 kB` |    <sup>0.1%↑</sup>`14.2 kB` |
| `dist/webpack/dispatch/Authentication.d.ts`         |   `3.2 kB` |       <sup>5%↑</sup>`3.3 kB` |
| `dist/webpack/dispatch/BasicAuthentication.d.ts`    |          — |                      `682 B` |
| `dist/webpack/dispatch/EnhancedAuthentication.d.ts` |   `3.2 kB` |    <sup>-79.4%↓</sup>`655 B` |
| **Total** <sub>_(Includes all files)_</sub>         |   `1.4 MB` |   <sup>-0.56%↓</sup>`1.3 MB` |
| **Tarball size**                                    | `107.4 kB` | <sup>-1.26%↓</sup>`106.0 kB` |

<details><summary>Unchanged files</summary>

| File                                                                                                   |       Size |
| ------------------------------------------------------------------------------------------------------ | ---------: |
| [`CHANGELOG.md`](https://github.com/williazz/aws-rum-web/blob/AuthRefactor/CHANGELOG.md)               |  `18.5 kB` |
| `dist/cjs/dispatch/BeaconHttpHandler.d.ts`                                                             |    `255 B` |
| `dist/cjs/dispatch/BeaconHttpHandler.js`                                                               |   `1.4 kB` |
| `dist/cjs/dispatch/CognitoIdentityClient.d.ts`                                                         |    `847 B` |
| `dist/cjs/dispatch/CognitoIdentityClient.js`                                                           |   `9.6 kB` |
| `dist/cjs/dispatch/dataplane.d.ts`                                                                     |    `631 B` |
| `dist/cjs/dispatch/dataplane.js`                                                                       |    `448 B` |
| `dist/cjs/dispatch/DataPlaneClient.d.ts`                                                               |    `802 B` |
| `dist/cjs/dispatch/DataPlaneClient.js`                                                                 |   `9.3 kB` |
| `dist/cjs/dispatch/Dispatch.d.ts`                                                                      |   `2.5 kB` |
| `dist/cjs/dispatch/Dispatch.js`                                                                        |  `10.8 kB` |
| `dist/cjs/dispatch/FetchHttpHandler.d.ts`                                                              |    `971 B` |
| `dist/cjs/dispatch/FetchHttpHandler.js`                                                                |   `4.2 kB` |
| `dist/cjs/dispatch/request-timeout.d.ts`                                                               |     `79 B` |
| `dist/cjs/dispatch/request-timeout.js`                                                                 |    `615 B` |
| `dist/cjs/dispatch/RetryHttpHandler.d.ts`                                                              |    `728 B` |
| `dist/cjs/dispatch/RetryHttpHandler.js`                                                                |   `5.3 kB` |
| `dist/cjs/dispatch/StsClient.d.ts`                                                                     |    `459 B` |
| `dist/cjs/dispatch/StsClient.js`                                                                       |   `6.0 kB` |
| `dist/cjs/dispatch/utils.d.ts`                                                                         |    `221 B` |
| `dist/cjs/dispatch/utils.js`                                                                           |   `3.6 kB` |
| `dist/cjs/errors/XhrError.d.ts`                                                                        |     `82 B` |
| `dist/cjs/errors/XhrError.js`                                                                          |   `1.2 kB` |
| `dist/cjs/event-bus/EventBus.d.ts`                                                                     |    `411 B` |
| `dist/cjs/event-bus/EventBus.js`                                                                       |   `1.5 kB` |
| `dist/cjs/event-cache/EventCache.d.ts`                                                                 |   `3.5 kB` |
| `dist/cjs/event-cache/EventCache.js`                                                                   |   `8.3 kB` |
| `dist/cjs/events/cumulative-layout-shift-event.d.ts`                                                   |   `1.0 kB` |
| `dist/cjs/events/cumulative-layout-shift-event.js`                                                     |    `312 B` |
| `dist/cjs/events/dom-event.d.ts`                                                                       |    `675 B` |
| `dist/cjs/events/dom-event.js`                                                                         |    `312 B` |
| `dist/cjs/events/first-input-delay-event.d.ts`                                                         |   `1.0 kB` |
| `dist/cjs/events/first-input-delay-event.js`                                                           |    `312 B` |
| `dist/cjs/events/http-event.d.ts`                                                                      |    `840 B` |
| `dist/cjs/events/http-event.js`                                                                        |    `312 B` |
| `dist/cjs/events/js-error-event.d.ts`                                                                  |    `503 B` |
| `dist/cjs/events/js-error-event.js`                                                                    |    `312 B` |
| `dist/cjs/events/largest-contentful-paint-event.d.ts`                                                  |   `1.4 kB` |
| `dist/cjs/events/largest-contentful-paint-event.js`                                                    |    `312 B` |
| `dist/cjs/events/meta-data.d.ts`                                                                       |    `833 B` |
| `dist/cjs/events/meta-data.js`                                                                         |    `312 B` |
| `dist/cjs/events/navigation-event.d.ts`                                                                |   `1.5 kB` |
| `dist/cjs/events/navigation-event.js`                                                                  |    `312 B` |
| `dist/cjs/events/page-view-event.d.ts`                                                                 |    `510 B` |
| `dist/cjs/events/page-view-event.js`                                                                   |    `312 B` |
| `dist/cjs/events/resource-event.d.ts`                                                                  |    `974 B` |
| `dist/cjs/events/resource-event.js`                                                                    |    `312 B` |
| `dist/cjs/events/time-to-interactive-event.d.ts`                                                       |    `387 B` |
| `dist/cjs/events/time-to-interactive-event.js`                                                         |    `312 B` |
| `dist/cjs/events/xray-trace-event.d.ts`                                                                |   `1.8 kB` |
| `dist/cjs/events/xray-trace-event.js`                                                                  |    `312 B` |
| `dist/cjs/index.d.ts`                                                                                  |    `894 B` |
| `dist/cjs/index.js`                                                                                    |   `2.1 kB` |
| `dist/cjs/orchestration/Orchestration.d.ts`                                                            |   `8.1 kB` |
| `dist/cjs/plugins/event-plugins/DomEventPlugin.d.ts`                                                   |   `1.5 kB` |
| `dist/cjs/plugins/event-plugins/DomEventPlugin.js`                                                     |   `7.0 kB` |
| `dist/cjs/plugins/event-plugins/FetchPlugin.d.ts`                                                      |   `1.2 kB` |
| `dist/cjs/plugins/event-plugins/FetchPlugin.js`                                                        |  `10.3 kB` |
| `dist/cjs/plugins/event-plugins/JsErrorPlugin.d.ts`                                                    |    `806 B` |
| `dist/cjs/plugins/event-plugins/JsErrorPlugin.js`                                                      |   `3.9 kB` |
| `dist/cjs/plugins/event-plugins/NavigationPlugin.d.ts`                                                 |   `2.3 kB` |
| `dist/cjs/plugins/event-plugins/NavigationPlugin.js`                                                   |  `12.9 kB` |
| `dist/cjs/plugins/event-plugins/PageViewPlugin.d.ts`                                                   |   `1.1 kB` |
| `dist/cjs/plugins/event-plugins/PageViewPlugin.js`                                                     |   `4.3 kB` |
| `dist/cjs/plugins/event-plugins/ResourcePlugin.d.ts`                                                   |    `828 B` |
| `dist/cjs/plugins/event-plugins/ResourcePlugin.js`                                                     |   `5.9 kB` |
| `dist/cjs/plugins/event-plugins/TTIPlugin.d.ts`                                                        |    `368 B` |
| `dist/cjs/plugins/event-plugins/TTIPlugin.js`                                                          |   `2.3 kB` |
| `dist/cjs/plugins/event-plugins/WebVitalsPlugin.d.ts`                                                  |    `491 B` |
| `dist/cjs/plugins/event-plugins/WebVitalsPlugin.js`                                                    |   `5.6 kB` |
| `dist/cjs/plugins/event-plugins/XhrPlugin.d.ts`                                                        |   `2.7 kB` |
| `dist/cjs/plugins/event-plugins/XhrPlugin.js`                                                          |  `13.0 kB` |
| `dist/cjs/plugins/InternalPlugin.d.ts`                                                                 |    `699 B` |
| `dist/cjs/plugins/InternalPlugin.js`                                                                   |    `888 B` |
| `dist/cjs/plugins/MonkeyPatched.d.ts`                                                                  |    `638 B` |
| `dist/cjs/plugins/MonkeyPatched.js`                                                                    |   `2.3 kB` |
| `dist/cjs/plugins/Plugin.d.ts`                                                                         |   `1.0 kB` |
| `dist/cjs/plugins/Plugin.js`                                                                           |     `77 B` |
| `dist/cjs/plugins/PluginManager.d.ts`                                                                  |   `1.3 kB` |
| `dist/cjs/plugins/PluginManager.js`                                                                    |   `2.9 kB` |
| `dist/cjs/plugins/types.d.ts`                                                                          |    `608 B` |
| `dist/cjs/plugins/types.js`                                                                            |     `77 B` |
| `dist/cjs/plugins/utils/constant.d.ts`                                                                 |    `778 B` |
| `dist/cjs/plugins/utils/constant.js`                                                                   |   `2.0 kB` |
| `dist/cjs/plugins/utils/http-utils.d.ts`                                                               |   `2.3 kB` |
| `dist/cjs/plugins/utils/http-utils.js`                                                                 |   `7.2 kB` |
| `dist/cjs/plugins/utils/js-error-utils.d.ts`                                                           |    `239 B` |
| `dist/cjs/plugins/utils/js-error-utils.js`                                                             |   `2.6 kB` |
| `dist/cjs/plugins/utils/performance-utils.d.ts`                                                        |    `723 B` |
| `dist/cjs/plugins/utils/performance-utils.js`                                                          |    `839 B` |
| `dist/cjs/sessions/PageManager.d.ts`                                                                   |   `2.4 kB` |
| `dist/cjs/sessions/PageManager.js`                                                                     |   `7.8 kB` |
| `dist/cjs/sessions/SessionManager.d.ts`                                                                |   `3.1 kB` |
| `dist/cjs/sessions/SessionManager.js`                                                                  |   `9.0 kB` |
| `dist/cjs/sessions/VirtualPageLoadTimer.d.ts`                                                          |   `2.8 kB` |
| `dist/cjs/sessions/VirtualPageLoadTimer.js`                                                            |   `8.6 kB` |
| `dist/cjs/time-to-interactive/QuietWindowSearch.d.ts`                                                  |   `1.0 kB` |
| `dist/cjs/time-to-interactive/QuietWindowSearch.js`                                                    |   `8.3 kB` |
| `dist/cjs/time-to-interactive/TimeToInteractive.d.ts`                                                  |   `1.7 kB` |
| `dist/cjs/time-to-interactive/TimeToInteractive.js`                                                    |   `1.9 kB` |
| `dist/cjs/time-to-interactive/VisuallyReadySearch.d.ts`                                                |    `768 B` |
| `dist/cjs/time-to-interactive/VisuallyReadySearch.js`                                                  |   `5.3 kB` |
| `dist/cjs/utils/common-utils.d.ts`                                                                     |   `3.2 kB` |
| `dist/cjs/utils/common-utils.js`                                                                       |   `6.4 kB` |
| `dist/cjs/utils/constants.d.ts`                                                                        |    `391 B` |
| `dist/cjs/utils/constants.js`                                                                          |    `594 B` |
| `dist/cjs/utils/cookies-utils.d.ts`                                                                    |   `1.0 kB` |
| `dist/cjs/utils/cookies-utils.js`                                                                      |   `2.3 kB` |
| `dist/cjs/utils/random.d.ts`                                                                           |     `74 B` |
| `dist/cjs/utils/random.js`                                                                             |    `420 B` |
| `dist/es/dispatch/BeaconHttpHandler.d.ts`                                                              |    `255 B` |
| `dist/es/dispatch/BeaconHttpHandler.js`                                                                |   `1.2 kB` |
| `dist/es/dispatch/CognitoIdentityClient.d.ts`                                                          |    `847 B` |
| `dist/es/dispatch/CognitoIdentityClient.js`                                                            |   `9.4 kB` |
| `dist/es/dispatch/dataplane.d.ts`                                                                      |    `631 B` |
| `dist/es/dispatch/dataplane.js`                                                                        |    `382 B` |
| `dist/es/dispatch/DataPlaneClient.d.ts`                                                                |    `802 B` |
| `dist/es/dispatch/DataPlaneClient.js`                                                                  |   `9.1 kB` |
| `dist/es/dispatch/Dispatch.d.ts`                                                                       |   `2.5 kB` |
| `dist/es/dispatch/Dispatch.js`                                                                         |  `10.6 kB` |
| `dist/es/dispatch/FetchHttpHandler.d.ts`                                                               |    `971 B` |
| `dist/es/dispatch/FetchHttpHandler.js`                                                                 |   `3.9 kB` |
| `dist/es/dispatch/request-timeout.d.ts`                                                                |     `79 B` |
| `dist/es/dispatch/request-timeout.js`                                                                  |    `471 B` |
| `dist/es/dispatch/RetryHttpHandler.d.ts`                                                               |    `728 B` |
| `dist/es/dispatch/RetryHttpHandler.js`                                                                 |   `5.1 kB` |
| `dist/es/dispatch/StsClient.d.ts`                                                                      |    `459 B` |
| `dist/es/dispatch/StsClient.js`                                                                        |   `5.9 kB` |
| `dist/es/dispatch/utils.d.ts`                                                                          |    `221 B` |
| `dist/es/dispatch/utils.js`                                                                            |   `3.4 kB` |
| `dist/es/errors/XhrError.d.ts`                                                                         |     `82 B` |
| `dist/es/errors/XhrError.js`                                                                           |   `1.1 kB` |
| `dist/es/event-bus/EventBus.d.ts`                                                                      |    `411 B` |
| `dist/es/event-bus/EventBus.js`                                                                        |   `1.4 kB` |
| `dist/es/event-cache/EventCache.d.ts`                                                                  |   `3.5 kB` |
| `dist/es/event-cache/EventCache.js`                                                                    |   `8.1 kB` |
| `dist/es/events/cumulative-layout-shift-event.d.ts`                                                    |   `1.0 kB` |
| `dist/es/events/cumulative-layout-shift-event.js`                                                      |    `246 B` |
| `dist/es/events/dom-event.d.ts`                                                                        |    `675 B` |
| `dist/es/events/dom-event.js`                                                                          |    `246 B` |
| `dist/es/events/first-input-delay-event.d.ts`                                                          |   `1.0 kB` |
| `dist/es/events/first-input-delay-event.js`                                                            |    `246 B` |
| `dist/es/events/http-event.d.ts`                                                                       |    `840 B` |
| `dist/es/events/http-event.js`                                                                         |    `246 B` |
| `dist/es/events/js-error-event.d.ts`                                                                   |    `503 B` |
| `dist/es/events/js-error-event.js`                                                                     |    `246 B` |
| `dist/es/events/largest-contentful-paint-event.d.ts`                                                   |   `1.4 kB` |
| `dist/es/events/largest-contentful-paint-event.js`                                                     |    `246 B` |
| `dist/es/events/meta-data.d.ts`                                                                        |    `833 B` |
| `dist/es/events/meta-data.js`                                                                          |    `246 B` |
| `dist/es/events/navigation-event.d.ts`                                                                 |   `1.5 kB` |
| `dist/es/events/navigation-event.js`                                                                   |    `246 B` |
| `dist/es/events/page-view-event.d.ts`                                                                  |    `510 B` |
| `dist/es/events/page-view-event.js`                                                                    |    `246 B` |
| `dist/es/events/resource-event.d.ts`                                                                   |    `974 B` |
| `dist/es/events/resource-event.js`                                                                     |    `246 B` |
| `dist/es/events/time-to-interactive-event.d.ts`                                                        |    `387 B` |
| `dist/es/events/time-to-interactive-event.js`                                                          |    `246 B` |
| `dist/es/events/xray-trace-event.d.ts`                                                                 |   `1.8 kB` |
| `dist/es/events/xray-trace-event.js`                                                                   |    `246 B` |
| `dist/es/index.d.ts`                                                                                   |    `894 B` |
| `dist/es/index.js`                                                                                     |    `611 B` |
| `dist/es/orchestration/Orchestration.d.ts`                                                             |   `8.1 kB` |
| `dist/es/plugins/event-plugins/DomEventPlugin.d.ts`                                                    |   `1.5 kB` |
| `dist/es/plugins/event-plugins/DomEventPlugin.js`                                                      |   `6.8 kB` |
| `dist/es/plugins/event-plugins/FetchPlugin.d.ts`                                                       |   `1.2 kB` |
| `dist/es/plugins/event-plugins/FetchPlugin.js`                                                         |  `10.0 kB` |
| `dist/es/plugins/event-plugins/JsErrorPlugin.d.ts`                                                     |    `806 B` |
| `dist/es/plugins/event-plugins/JsErrorPlugin.js`                                                       |   `3.7 kB` |
| `dist/es/plugins/event-plugins/NavigationPlugin.d.ts`                                                  |   `2.3 kB` |
| `dist/es/plugins/event-plugins/NavigationPlugin.js`                                                    |  `12.7 kB` |
| `dist/es/plugins/event-plugins/PageViewPlugin.d.ts`                                                    |   `1.1 kB` |
| `dist/es/plugins/event-plugins/PageViewPlugin.js`                                                      |   `4.1 kB` |
| `dist/es/plugins/event-plugins/ResourcePlugin.d.ts`                                                    |    `828 B` |
| `dist/es/plugins/event-plugins/ResourcePlugin.js`                                                      |   `5.7 kB` |
| `dist/es/plugins/event-plugins/TTIPlugin.d.ts`                                                         |    `368 B` |
| `dist/es/plugins/event-plugins/TTIPlugin.js`                                                           |   `2.1 kB` |
| `dist/es/plugins/event-plugins/WebVitalsPlugin.d.ts`                                                   |    `491 B` |
| `dist/es/plugins/event-plugins/WebVitalsPlugin.js`                                                     |   `5.3 kB` |
| `dist/es/plugins/event-plugins/XhrPlugin.d.ts`                                                         |   `2.7 kB` |
| `dist/es/plugins/event-plugins/XhrPlugin.js`                                                           |  `12.7 kB` |
| `dist/es/plugins/InternalPlugin.d.ts`                                                                  |    `699 B` |
| `dist/es/plugins/InternalPlugin.js`                                                                    |    `758 B` |
| `dist/es/plugins/MonkeyPatched.d.ts`                                                                   |    `638 B` |
| `dist/es/plugins/MonkeyPatched.js`                                                                     |   `2.2 kB` |
| `dist/es/plugins/Plugin.d.ts`                                                                          |   `1.0 kB` |
| `dist/es/plugins/Plugin.js`                                                                            |     `11 B` |
| `dist/es/plugins/PluginManager.d.ts`                                                                   |   `1.3 kB` |
| `dist/es/plugins/PluginManager.js`                                                                     |   `2.7 kB` |
| `dist/es/plugins/types.d.ts`                                                                           |    `608 B` |
| `dist/es/plugins/types.js`                                                                             |     `11 B` |
| `dist/es/plugins/utils/constant.d.ts`                                                                  |    `778 B` |
| `dist/es/plugins/utils/constant.js`                                                                    |   `1.4 kB` |
| `dist/es/plugins/utils/http-utils.d.ts`                                                                |   `2.3 kB` |
| `dist/es/plugins/utils/http-utils.js`                                                                  |   `6.0 kB` |
| `dist/es/plugins/utils/js-error-utils.d.ts`                                                            |    `239 B` |
| `dist/es/plugins/utils/js-error-utils.js`                                                              |   `2.4 kB` |
| `dist/es/plugins/utils/performance-utils.d.ts`                                                         |    `723 B` |
| `dist/es/plugins/utils/performance-utils.js`                                                           |    `561 B` |
| `dist/es/sessions/PageManager.d.ts`                                                                    |   `2.4 kB` |
| `dist/es/sessions/PageManager.js`                                                                      |   `7.7 kB` |
| `dist/es/sessions/SessionManager.d.ts`                                                                 |   `3.1 kB` |
| `dist/es/sessions/SessionManager.js`                                                                   |   `8.5 kB` |
| `dist/es/sessions/VirtualPageLoadTimer.d.ts`                                                           |   `2.8 kB` |
| `dist/es/sessions/VirtualPageLoadTimer.js`                                                             |   `8.4 kB` |
| `dist/es/time-to-interactive/QuietWindowSearch.d.ts`                                                   |   `1.0 kB` |
| `dist/es/time-to-interactive/QuietWindowSearch.js`                                                     |   `8.2 kB` |
| `dist/es/time-to-interactive/TimeToInteractive.d.ts`                                                   |   `1.7 kB` |
| `dist/es/time-to-interactive/TimeToInteractive.js`                                                     |   `1.7 kB` |
| `dist/es/time-to-interactive/VisuallyReadySearch.d.ts`                                                 |    `768 B` |
| `dist/es/time-to-interactive/VisuallyReadySearch.js`                                                   |   `5.1 kB` |
| `dist/es/utils/common-utils.d.ts`                                                                      |   `3.2 kB` |
| `dist/es/utils/common-utils.js`                                                                        |   `5.9 kB` |
| `dist/es/utils/constants.d.ts`                                                                         |    `391 B` |
| `dist/es/utils/constants.js`                                                                           |    `326 B` |
| `dist/es/utils/cookies-utils.d.ts`                                                                     |   `1.0 kB` |
| `dist/es/utils/cookies-utils.js`                                                                       |   `2.0 kB` |
| `dist/es/utils/random.d.ts`                                                                            |     `74 B` |
| `dist/es/utils/random.js`                                                                              |    `273 B` |
| `dist/webpack/CommandQueue.d.ts`                                                                       |   `1.3 kB` |
| `dist/webpack/dispatch/BeaconHttpHandler.d.ts`                                                         |    `255 B` |
| `dist/webpack/dispatch/CognitoIdentityClient.d.ts`                                                     |    `847 B` |
| `dist/webpack/dispatch/dataplane.d.ts`                                                                 |    `631 B` |
| `dist/webpack/dispatch/DataPlaneClient.d.ts`                                                           |    `802 B` |
| `dist/webpack/dispatch/Dispatch.d.ts`                                                                  |   `2.5 kB` |
| `dist/webpack/dispatch/FetchHttpHandler.d.ts`                                                          |    `971 B` |
| `dist/webpack/dispatch/request-timeout.d.ts`                                                           |     `79 B` |
| `dist/webpack/dispatch/RetryHttpHandler.d.ts`                                                          |    `728 B` |
| `dist/webpack/dispatch/StsClient.d.ts`                                                                 |    `459 B` |
| `dist/webpack/dispatch/utils.d.ts`                                                                     |    `221 B` |
| `dist/webpack/errors/XhrError.d.ts`                                                                    |     `82 B` |
| `dist/webpack/event-bus/EventBus.d.ts`                                                                 |    `411 B` |
| `dist/webpack/event-cache/EventCache.d.ts`                                                             |   `3.5 kB` |
| `dist/webpack/events/cumulative-layout-shift-event.d.ts`                                               |   `1.0 kB` |
| `dist/webpack/events/dom-event.d.ts`                                                                   |    `675 B` |
| `dist/webpack/events/first-input-delay-event.d.ts`                                                     |   `1.0 kB` |
| `dist/webpack/events/http-event.d.ts`                                                                  |    `840 B` |
| `dist/webpack/events/js-error-event.d.ts`                                                              |    `503 B` |
| `dist/webpack/events/largest-contentful-paint-event.d.ts`                                              |   `1.4 kB` |
| `dist/webpack/events/meta-data.d.ts`                                                                   |    `833 B` |
| `dist/webpack/events/navigation-event.d.ts`                                                            |   `1.5 kB` |
| `dist/webpack/events/page-view-event.d.ts`                                                             |    `510 B` |
| `dist/webpack/events/resource-event.d.ts`                                                              |    `974 B` |
| `dist/webpack/events/xray-trace-event.d.ts`                                                            |   `1.8 kB` |
| `dist/webpack/index-browser.d.ts`                                                                      |    `184 B` |
| `dist/webpack/orchestration/Orchestration.d.ts`                                                        |   `8.1 kB` |
| `dist/webpack/plugins/event-plugins/DomEventPlugin.d.ts`                                               |   `1.5 kB` |
| `dist/webpack/plugins/event-plugins/FetchPlugin.d.ts`                                                  |   `1.2 kB` |
| `dist/webpack/plugins/event-plugins/JsErrorPlugin.d.ts`                                                |    `806 B` |
| `dist/webpack/plugins/event-plugins/NavigationPlugin.d.ts`                                             |   `2.3 kB` |
| `dist/webpack/plugins/event-plugins/PageViewPlugin.d.ts`                                               |   `1.1 kB` |
| `dist/webpack/plugins/event-plugins/ResourcePlugin.d.ts`                                               |    `828 B` |
| `dist/webpack/plugins/event-plugins/WebVitalsPlugin.d.ts`                                              |    `491 B` |
| `dist/webpack/plugins/event-plugins/XhrPlugin.d.ts`                                                    |   `2.7 kB` |
| `dist/webpack/plugins/InternalPlugin.d.ts`                                                             |    `699 B` |
| `dist/webpack/plugins/MonkeyPatched.d.ts`                                                              |    `638 B` |
| `dist/webpack/plugins/Plugin.d.ts`                                                                     |   `1.0 kB` |
| `dist/webpack/plugins/PluginManager.d.ts`                                                              |   `1.3 kB` |
| `dist/webpack/plugins/types.d.ts`                                                                      |    `608 B` |
| `dist/webpack/plugins/utils/constant.d.ts`                                                             |    `778 B` |
| `dist/webpack/plugins/utils/http-utils.d.ts`                                                           |   `2.3 kB` |
| `dist/webpack/plugins/utils/js-error-utils.d.ts`                                                       |    `239 B` |
| `dist/webpack/plugins/utils/performance-utils.d.ts`                                                    |    `723 B` |
| `dist/webpack/remote-config/remote-config.d.ts`                                                        |    `510 B` |
| `dist/webpack/sessions/PageManager.d.ts`                                                               |   `2.4 kB` |
| `dist/webpack/sessions/SessionManager.d.ts`                                                            |   `3.1 kB` |
| `dist/webpack/sessions/VirtualPageLoadTimer.d.ts`                                                      |   `2.8 kB` |
| `dist/webpack/utils/common-utils.d.ts`                                                                 |   `3.2 kB` |
| `dist/webpack/utils/constants.d.ts`                                                                    |    `391 B` |
| `dist/webpack/utils/cookies-utils.d.ts`                                                                |   `1.0 kB` |
| `dist/webpack/utils/random.d.ts`                                                                       |     `74 B` |
| [`LICENSE`](https://github.com/williazz/aws-rum-web/blob/AuthRefactor/LICENSE)                         |  `10.1 kB` |
| [`LICENSE-THIRD-PARTY`](https://github.com/williazz/aws-rum-web/blob/AuthRefactor/LICENSE-THIRD-PARTY) | `632.4 kB` |
| [`NOTICE`](https://github.com/williazz/aws-rum-web/blob/AuthRefactor/NOTICE)                           |     `67 B` |
| [`package.json`](https://github.com/williazz/aws-rum-web/blob/AuthRefactor/package.json)               |   `6.5 kB` |
| [`README.md`](https://github.com/williazz/aws-rum-web/blob/AuthRefactor/README.md)                     |   `4.0 kB` |
</details>

<sub>🤖 This report was automatically generated by [pkg-size-action](https://github.com/pkg-size/action/)</sub>

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by williazz** [comment]
For "deprecate Credentials for AwsCredentialIdentity", this is a cosmetic change. Credentials extends AwsCredentialIdentity, but does not add any fields. So they are the same interface

**Review by qhanam — COMMENTED** [review]
Nice thanks! This is good maintainability improvement and saves a few bytes.

One blocking issue (re-enable the unbound-method check) and a few nits.

File: .eslintrc.js [hunk 1]
```diff
@@ -237,6 +237,7 @@ module.exports = {
             }
         ],
         'use-isnan': 'error',
-        'valid-typeof': 'off'
+        'valid-typeof': 'off',
+        '@typescript-eslint/unbound-method': 'off'
```
[the remark this answers is not in the document]
Comment (williazz) [reply .eslintrc.js]: Restored

[the remark this answers is not in the document]
Comment (williazz) [reply .eslintrc.js]: I noticed that we do disable this rule quite a lot already, but yeah happy to remove from this PR.

**Review by qhanam — APPROVED** [review]

**Pull request merged**